dejavu/__init__.py

```diff
@@ -122,45 +124,110 @@ def align_matches(self, matches):
             consensus about which hashes are "true" signal from the audio.
 
             Returns a dictionary with match information.
+
+            If "multiple_match" is set to True in the config, this will return
+            a list of dictionaries with match information.
         """
         # align by diffs
         diff_counter = {}
-        largest = 0
-        largest_count = 0
-        song_id = -1
-        for tup in matches:
-            sid, diff = tup
-            if diff not in diff_counter:
-                diff_counter[diff] = {}
-            if sid not in diff_counter[diff]:
-                diff_counter[diff][sid] = 0
-            diff_counter[diff][sid] += 1
-
-            if diff_counter[diff][sid] > largest_count:
-                largest = diff
-                largest_count = diff_counter[diff][sid]
-                song_id = sid
-
-        # extract idenfication
-        song = self.db.get_song_by_id(song_id)
-        if song:
-            # TODO: Clarify what `get_song_by_id` should return.
-            songname = song.get(Dejavu.SONG_NAME, None)
+
+        if self.multiple_match:
```

Comment:
this could be two different functions -- it's pretty long and unwieldy. Not a big deal though